generatedocs.py

```diff
@@ -0,0 +1,413 @@
+from git import Repo # git for downloading assets
+from jinja2 import Environment, FileSystemLoader # template magic
+import json # reading the input file
+import os # file paths
+import re # regex for searching for assets files
+from tqdm import tqdm # progress bar
+
+################################################################################
+#                      ASSET COMPONENTS HELPER FUNCTIONS                       #
+################################################################################
+
+# Clones asset directory from OpenSpace git repository
+# Uses the latest master
+# Returns absolute path to the asset folder
+def cloneAssetsFolderGit(folderName):
+    def printProgress(op_code, cur_count, max_count=None, message=''):
+        print(message)
+
+    print("Cloning assets examples...")
+    repo = Repo.init(folderName)
+    branchName = "origin/master"
+
+    # Create a new remote if there isn't one already created
+    origin = repo.remotes[0] if len(repo.remotes) > 0 else None 
+    if not origin:
+        print("No remote origin found. Creating OpenSpace remote...")
+        origin = repo.create_remote("origin", "https://github.com/OpenSpace/OpenSpace")
+
+    print("Fetching OpenSpace on branch {}... this might take a while".format(branchName))
+    origin.fetch(progress=printProgress)
+
+    dataAssetsPath = "data/assets" 
+    git = repo.git()
+    git.checkout(branchName, "--", dataAssetsPath)
+    print("Done cloning assets folder from OpenSpace repository")
+    assetsFolderPath = os.path.abspath(os.path.join(folderName, dataAssetsPath))
+    return assetsFolderPath
+
+# Returns the number of lines in a file
+def getFileLength(path):
+    try:
+        # Read binary file as it is faster and we only want to know the length
+        with open(path, 'rb') as fp:
+            if fp.readable():
+                length = len(fp.read()) 
+                return length
+            else:
+                return None
+    except IOError:
+        input("Could not open file path ", path)
+        return None
+
+# Find all .asset files in the root dir and subdirs
+def assetsInPathRecursive(root):
+    filenames = []
+
+    for path, subdirs, files in os.walk(root):
+        for file in files:
+            if file.endswith(".asset"):
+                filenames.append(os.path.join(path, file))
+
+    return filenames 
```

Comment:
I suggest no blank lines in a short function like this. The python style guide recommends keeping spaces within a function to a minimum, or you could also have two spaces between functions that are outside of a class.